### PR TITLE
First Pass - Amp Subsystem

### LINKED ISSUE #6: First Pass - Amp Subsystem
Functionality Expected:

- Amp Subsystem
- Amp Eject Command

### PR DESCRIPTION
**Pull request by AgrawalDevansh09**

### CODE REVIEW

File: src/main/cpp/RobotContainer.cpp [hunk 1]
```diff
@@ -3,15 +3,20 @@
...
 #include "RobotContainer.h"
-
+#include "Commands/AmpCommands.h"
 #include <frc2/command/Commands.h>
 
+
+
 RobotContainer::RobotContainer() {
-  ConfigureBindings();
+  DriverController.X().OnTrue(AmpEject(&Amp));
```
Comment (YeetDot) [inline src/main/cpp/RobotContainer.cpp]: These should not be in the Constructor, but rather in ConfigureBindings.

File: src/main/include/RobotContainer.h [hunk 2]
```diff
@@ -6,12 +6,24 @@
 
 #include <frc2/command/CommandPtr.h>
 
+#include "Subsystems/AmpSubsystem.h"
+#include <frc/XboxController.h>
```
Comment (YeetDot) [inline src/main/include/RobotContainer.h]: Are all these includes required?

**Review by YeetDot — CHANGES_REQUESTED** [review]
Change those things

**Comment by AgrawalDevansh09** [comment]
Implimented Changes

**Review by YeetDot — APPROVED** [review]
I think it's good now but <sub>Camden</sub> probably has something to say

File: src/main/cpp/Subsystems/AmpSubsystem.cpp [hunk 3]
```diff
@@ -0,0 +1,22 @@
+#include "Subsystems/AmpSubsystem.h"
+
+AmpSubsystem::AmpSubsystem()
+{
+    AmpMotor.SetSmartCurrentLimit(40);
+}
+
+void AmpSubsystem::AmpExtend(double AmpExtendPower)
```
Comment (cpastyrn3357) [inline src/main/cpp/Subsystems/AmpSubsystem.cpp]: Probably remove amp extend. Not needed and could cause confusion

File: src/main/include/Subsystems/AmpSubsystem.h [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+#include <frc2/command/CommandPtr.h>
+#include <frc2/command/SubsystemBase.h>
+#include <rev/CANSparkMax.h>
+
+class AmpSubsystem : public frc2::SubsystemBase
+{
+    public:
+        AmpSubsystem();
+        //void Periodic() override;
+        void AmpExtend(double AmpExtendPower);
```
Comment (cpastyrn3357) [inline src/main/include/Subsystems/AmpSubsystem.h]: And remove amp extend here too.

File: src/main/cpp/RobotContainer.cpp [hunk 5]
```diff
@@ -16,8 +16,12 @@ void RobotContainer::ConfigureBindings() {
...
   driverController.B().WhileTrue(IndexerEject(&indexer));
   driverController.B().OnFalse(IndexerStop(&indexer));
+  driverController.X().OnTrue(AmpEject(&Amp));
+  driverController.X().OnFalse(AmpStop(&Amp));
 }
 
 frc2::CommandPtr RobotContainer::GetAutonomousCommand() {
   return frc2::cmd::Print("No autonomous command configured");
 }
+
```
Comment (cpastyrn3357) [inline src/main/cpp/RobotContainer.cpp:26]: Nit picky, but try to remove changes in whitespace where possible

File: src/main/include/Subsystems/AmpSubsystem.h [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+#include <frc2/command/CommandPtr.h>
+#include <frc2/command/SubsystemBase.h>
+#include <rev/CANSparkMax.h>
+
+class AmpSubsystem : public frc2::SubsystemBase
+{
+    public:
+        AmpSubsystem();
+        //void Periodic() override;
```
Comment (cpastyrn3357) [inline src/main/include/Subsystems/AmpSubsystem.h]: Remove periodic, since it is not being used.

File: src/main/include/Commands/AmpCommands.h [hunk 7]
```diff
@@ -0,0 +1,8 @@
+#pragma once
+
+#include <frc2/command/CommandPtr.h>
+#include "Subsystems/AmpSubsystem.h"
+#include "frc2/command/Commands.h"
```
Comment (cpastyrn3357) [inline src/main/include/Commands/AmpCommands.h]: Move this include to the cpp file and use < > for WPILib includes

**Review by cpastyrn3357 — CHANGES_REQUESTED** [review]

**Review by cpastyrn3357 — APPROVED** [review]

**Comment by AgrawalDevansh09** [comment]
Merged into Amp

**Pull request merged**